test/app-tap/module_api.c

```diff
@@ -1,3 +1,10 @@
+/*
+ * Use assert() as a testing assertion function disregarding
+ * tarantool's build type.
+ */
+#undef NDEBUG
```

Comment:
I'm not against it, but this looks weird. Maybe, replace the assertions with `fail_if` from `unit.h` instead?

BTW personally I'm quite used to void-ing asserted variables (we do it all the time in the Tarantool source code). Lack of this `void` casts looks uneasy to me.

Reply:
No problem, I'll add `fail_if` macro into the file and will replace `assert()`s with it.

Reply:
Replaced `assert()` calls with `fail_unless()`.